OpenRA.Mods.Common/Traits/PaletteEffects/FlashPaletteEffect.cs

```diff
@@ -0,0 +1,83 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Drawing;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	using GUtil = OpenRA.Graphics.Util;
+
+	[Desc("Used for bursted one-colored whole screen effects. Add this to the world actor.")]
+	public class FlashPaletteEffectInfo : ITraitInfo
+	{
+		public readonly string[] ExcludePalettes = { "cursor", "chrome", "colorpicker", "fog", "shroud" };
+
+		[Desc("Measured in ticks.")]
+		public readonly int Length = 20;
+
+		public readonly Color Color = Color.White;
+
+		public object Create(ActorInitializer init) { return new FlashPaletteEffect(this); }
+	}
+
+	public class FlashPaletteEffect : IPaletteModifier, ITick
+	{
+		readonly FlashPaletteEffectInfo info;
+		bool flashing;
+
+		public FlashPaletteEffect(FlashPaletteEffectInfo info)
+		{
+			this.info = info;
+		}
+
+		int remainingFrames;
+
+		public void Enable(int ticks = -1)
+		{
+			if (ticks == -1)
+				remainingFrames = info.Length;
+			else
+				remainingFrames = ticks;
+		}
+
+		public void Tick(Actor self)
+		{
+			if (remainingFrames > 0)
+				remainingFrames--;
+		}
+
+		public void AdjustPalette(IReadOnlyDictionary<string, MutablePalette> palettes)
+		{
+			if (remainingFrames == 0 || flashing)
+				return;
+
+			flashing = true;
```

Comment:
I suggest you take a closer look at this code.  As it is written, this doesn't do anything.